include/uk/plat/irq.h

```diff
@@ -56,6 +58,13 @@ typedef int (*irq_handler_func_t)(void *);
  */
 int ukplat_irq_register(unsigned long irq, irq_handler_func_t func, void *arg);
 
+struct ukplat_event_irq_data {
+	struct __regs *regs;
+	unsigned long irq;
+};
+
+#define UKARCH_EVENT_IRQ event_irq
```

Comment:
We should call it `UKPLAT_` since the ctx structure is also `ukplat` and it is accessible via the `plat` header. Putting in it the `arch/x86/x86_64/include/uk/asm/lcpu.h` header (and duplicate for other architectures) would also work with `ukarch` prefix. However, we are currently discussing that the `arch` folder should only contain things actually defined by the architectural manual. And the events are more of a platform construct anyways.

In any case also put the prefix in the actual event name. We should have done this also for the trap events. I'll probably change it during re-arch. 

Also add a remark when this event is raised.
```suggestion
/** Bla */
#define UKPLAT_EVENT_IRQ ukplat_event_irq
```